scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,31 @@
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+from libc cimport math
+
+
+ctypedef fused real:
+    short
+    int
+    long
+    float
+    double
```

Comment:
Is there a list of numpy dtypes somewhere? I'll generate the full cross-product of possible signatures (http://docs.cython.org/src/userguide/fusedtypes.html#fused-types-and-arrays) and include "object" at the end, so that all use-cases (... even the nonsensical ones) are covered (To be honest I think we should also deprecate passing non-real inputs in).